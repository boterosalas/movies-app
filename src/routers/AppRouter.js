import React, { useState } from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route
} from "react-router-dom";
import LoginScreen from '../components/login/LoginScreen';
import DashboardRoute from './DashboardRoute';
import { UserContext } from './UserContext';


const AppRouter = () => {
    const [user, setUser] = useState(null);
    return (
        <UserContext.Provider value={{ user, setUser }} >
            <Router>
                <div>
                    <Switch>
                        <Route exac path="/login" component={LoginScreen} />
                        <Route exac path="/" component={DashboardRoute} />
                    </Switch>
                </div>
            </Router>
        </UserContext.Provider>
    )
}

export default AppRouter
