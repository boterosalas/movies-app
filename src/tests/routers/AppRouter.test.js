import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import AppRouter from '../../routers/AppRouter';


describe('Test on <AppRouter />',()=>{

    test('Should display the component correctly', () => {
        const wrapper = shallow(<AppRouter />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('BrowserRouter').exists()).toBe(true);
        expect(wrapper.find('Route').length).toBe(2);
    });
    
})