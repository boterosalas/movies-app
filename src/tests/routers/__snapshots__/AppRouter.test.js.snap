// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <AppRouter /> Should display the component correctly 1`] = `
<ContextProvider
  value={
    Object {
      "setUser": [Function],
      "user": null,
    }
  }
>
  <BrowserRouter>
    <div>
      <Switch>
        <Route
          component={[Function]}
          exac={true}
          path="/login"
        />
        <Route
          component={[Function]}
          exac={true}
          path="/"
        />
      </Switch>
    </div>
  </BrowserRouter>
</ContextProvider>
`;
